ansible_base/resource_registry/utils/sync_to_resource_server.py

```diff
@@ -39,29 +39,32 @@ def sync_to_resource_server(instance, action, ansible_id=None):
         logger.info(f"Skipping sync of resource {instance}")
         return
 
+    if action != "delete" and ansible_id is not None:
+        raise Exception("ansible_id should not be provided for create/update actions")
+    elif action == "delete" and ansible_id is None:
+        raise Exception("ansible_id should be provided for delete actions")
+
     try:
-        if action != "delete" and ansible_id is not None:
-            raise Exception("ansible_id should not be provided for create/update actions")
-        elif action == "delete" and ansible_id is None:
-            raise Exception("ansible_id should be provided for delete actions")
-        elif not getattr(instance, 'resource', None) or not instance.resource.ansible_id:
-            # We can't sync if we don't have a resource and an ansible_id.
-            logger.error(f"Resource {instance} does not have a resource or ansible_id")
-            return
+        resource = Resource.get_resource_for_object(instance)
     except Resource.DoesNotExist:
-        # The getattr() will raise a Resource.DoesNotExist if the resource doesn't exist.
         logger.error(f"Resource {instance} does not have a resource")
         return
 
+    if str(resource.service_id) == service_id() and action == "update":
+        # Don't sync if we're updating a resource that isn't owned by the resource server yet.
```

Comment:
> resource that isn't owned by the resource server

this seems to be opposite of

> `str(resource.service_id) == service_id()`

If we're here, doesn't that mean that it _is_ local?

Reply:
Yes, and we're bailing out (returning) in that case.

If it is local, and it's an update, we don't sync. We wait until it is owned by the resource server before we allow syncing updates. At least that is my understanding of the requirement. @newswangerd

Reply:
so... it's only local in the case that the creation signal went wrong, or the server is migrated and the data hasn't yet been migrated?

Reply:
@AlanCoding that's correct.